### DISCUSSION TITLE
transpile node_modules on server side

### DISCUSSION THREAD

**Discussion by gaurav5430** [vercel/next.js#26816] [Help]
### What version of Next.js are you using?

10.0.7

### What version of Node.js are you using?

12.18.3

### What browser are you using?

Chrome

### What operating system are you using?

Mac

### How are you deploying your application?

AWS

### Describe the Bug

extending my babel.config.js with next/babel , and extending my webpack config to include a package from node_modules, does not transpile the package on the server side / node.

I see that there is a solution to use next-transpile-modules, but I am just not sure why the current setup would not work.
I use the setup in other browser only projects to transpile a particular node module through webpack / babel config. 

### Expected Behavior

I would expect the babel loader to correctly transpile the node_modules package for usage in server / node.js, since i am using next/babel

### To Reproduce

```
// babel config.js

module.exports = {
  presets: [
    [
      'next/babel',
      {
        useBuiltIns: 'entry',
        corejs: '3',
      },
    ],
    '@babel/preset-typescript',
    '@babel/preset-react',
  ],
  plugins: [
    [
      'react-css-modules',
      {
        autoResolveMultipleImports: true,
        generateScopedName: '[name]__[local]__[hash:base64:5]',
        handleMissingStyleName: 'warn',
        webpackHotModuleReloading: true,
      },
    ],
    '@babel/plugin-syntax-dynamic-import',
    '@babel/plugin-proposal-class-properties',
    '@babel/plugin-proposal-object-rest-spread',
    '@babel/plugin-proposal-optional-chaining',
    '@babel/plugin-proposal-nullish-coalescing-operator',
  ],
};

```

custom webpack babel loader config:
```
rules: [
      {
        test: /\.(js|tsx)$/,
        include: [
          path.resolve(__dirname, 'node_modules/@packageToInclude/'),
          path.resolve(__dirname, '/src'),
        ],
        use: {
          loader: 'babel-loader',
        },
      },
...
...
```

**Comment by jamsinclair** [comment]
@gaurav5430 this is to be expected. You'll want to look into the third party library https://github.com/martpie/next-transpile-modules. Best of luck!

**Comment by gaurav5430** [comment]
why is this expected?

i have been depending on that library till now, but due to recent changes it has become difficult for me to continue using that library

which is why I am just trying to understand why it can't work like the above config

**Comment by jamsinclair** [comment]
I'm no Next.js expert 😅 , but it heavily configures the webpack config to help build the optimised bundles. You'll need to override or augment these existing configs to support your use case.

You can see that in the [source code for the next-transpile-modules library](https://github.com/martpie/next-transpile-modules/blob/master/src/next-transpile-modules.js).

Likely you'll need to make use of the internal next babel loader and play with the webpack externals config. Which will start to get quite hacky.

> due to recent changes it has become difficult for me to continue using that library

The author seems to be doing great work, if you're having issues might be worth starting an issue there! 😸

**Reply by gaurav5430** [reply]
hey, yeah the library is great, just that their module resolution approach has changed recently, which is causing a lot of friction in our setup. May be we will have a better pattern to still be able to use next-transpile-modules, but i was just trying to understand if there is something that we can do on our own, so as to not depend on future changes in the library. We are not trying to solve all of the problems that the existing library solves, just our own small problem.
